frontend/catalyst/jax_tracer.py

```diff
@@ -827,7 +828,7 @@ def trace_quantum_function(
 
     with EvaluationContext(EvaluationMode.QUANTUM_COMPILATION) as ctx:
         # (1) - Classical tracing
-        quantum_tape = QuantumTape()
+        quantum_tape = QuantumTape(shots=device.shots)
```

Comment:
Is this a bug we've had?

Reply:
This is the only question I have, were we missing this before?

Reply:
This is only for the new device API, we set shots on tape because potentially some transforms can split the shots on multiple tapes.